netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,168 @@
+    override def mouseClicked(e: MouseEvent): Unit = {
+      usageBox.setVisible(false)
+      val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+      dataModel.synchronized(dataModel.setRowCount(0))
+      editorArea.select(token.start, token.end)
+    }
+  })
+
+  usageTable.addKeyListener(new KeyAdapter(){
+    override def keyPressed(e: KeyEvent): Unit = {
+      e.getKeyCode match {
+        case KeyEvent.VK_ESCAPE => usageBox.setVisible(false)
+        case _ => e.setSource(editorArea)
+          editorArea.dispatchEvent(e)
+      }
+    }
+  })
+
+  usageTable.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR))
+
+  def showBox(me: MouseEvent, position: Int): Unit = {
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), position)
+    for {token <- tokenOption} {
+      if(token.tpe == TokenType.Ident || token.tpe == TokenType.Command || token.tpe == TokenType.Reporter) {
+        val tokens = getUsage(editorArea.getText(), token)
+        dataModel.setRowCount(0)
+        for (t <- tokens) {
+          dataModel.addRow(Array[AnyRef](t, editorArea.getLineText(t.start).trim))
+        }
+        if (dataModel.getRowCount != 0) {
+          usageTable.setDefaultRenderer(classOf[String], new LineRenderer(Some(token.text)))
+          usageTable.setPreferredScrollableViewportSize(usageTable.getPreferredSize())
+          usageTable.setFillsViewportHeight(true)
+          usageBox.setSize(usageTable.getPreferredSize)
+          usageTable.validate()
+          usageBox.setLocation(me.getLocationOnScreen)
+          usageBox.setVisible(true)
+        }
+      }
+    }
+  }
+
+  def getUsage(source: String, token: Token): Iterator[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.filter(t => t.text.equalsIgnoreCase(token.text) && t.start != token.start)
+  }
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  class LineRenderer(boldedString: Option[String]) extends TableCellRenderer {
+    override def getTableCellRendererComponent(table: JTable, value: AnyRef, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component = {
+      val pane  = new JEditorPane()
+      pane.setOpaque(true)
+      pane.setBorder(BorderFactory.createEmptyBorder(1,0,0,0))
+      val editorKit = boldedString match {
+        case None => new HighlightEditorKit(null, editorArea.colorizer)
+        case Some(selectedString) => new BoldEditorKit(selectedString)
+      }
+      pane.setEditorKit(editorKit)
+      pane.setText(value.asInstanceOf[String])
+      val alternate = UIManager.getColor("Table.alternateRowColor")
+      val defaults = new UIDefaults()
+      if (row%2 == 0){
```

Comment:
That's correct. Both of these colors are the same on Mac OS X, so I assumed we didn't have it in this case.